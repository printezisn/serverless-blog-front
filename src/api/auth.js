import { CognitoAuth } from "amazon-cognito-auth-js"
import { Events } from "../utils/constants"
import { eventBus } from "../utils/eventBus"

class Auth {
    constructor() {
        const rootUrl = process.env.GATSBY_EDNABLOG_SITE_ROOT_URL
        const authData = {
            ClientId: "7icdk42lkcj7fevgp2lgss0aio",
            AppWebDomain: "ednablog.auth.ca-central-1.amazoncognito.com",
            TokenScopesArray: ["email", "openid"],
            RedirectUriSignIn: rootUrl,
            RedirectUriSignOut: rootUrl,
            UserPoolId: "ca-central-1_zPiF08WsY",
            AdvancedSecurityDataCollectionFlag: false,
        }

        const self = this
        const auth = new CognitoAuth(authData)
        auth.userhandler = {
            onSuccess: result => {
                self.accessToken = result.getAccessToken().getJwtToken()
                eventBus.emit(Events.AUTH_CHANGE)
            },
            onFailure: () => {},
        }

        this.auth = auth
        this.accessToken = this.auth
            .getSignInUserSession()
            .getAccessToken()
            .getJwtToken()
    }

    logIn() {
        this.auth.getSession()
    }

    logOut() {
        this.auth.signOut()
    }

    init(url) {
        this.auth.parseCognitoWebResponse(url)
    }

    getAccessToken() {
        return this.accessToken
    }

    isLoggedIn() {
        return this.accessToken !== ""
    }
}

export const auth = new Auth()
