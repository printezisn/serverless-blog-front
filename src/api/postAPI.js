class PostAPI {
    constructor() {
        this.root = process.env.GATSBY_EDNABLOG_API_ROOT_URL
    }

    async createPost(post) {
        const response = await fetch(this.root, {
            method: "PUT",
            mode: "cors",
            body: JSON.stringify(post),
        })

        return response.json()
    }

    async updatePost(post) {
        const response = await fetch(this.root, {
            method: "POST",
            mode: "cors",
            body: JSON.stringify(post),
        })

        return response.json()
    }

    async getPost(id) {
        const response = await fetch(this.root + encodeURIComponent(id), {
            method: "GET",
            mode: "cors",
        })

        return response.json()
    }

    async deletePost(id) {
        const response = await fetch(this.root + encodeURIComponent(id), {
            method: "DELETE",
            mode: "cors",
        })

        return response.json()
    }
}

export const postAPI = new PostAPI()
