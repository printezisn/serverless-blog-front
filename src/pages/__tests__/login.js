import React from "react"
import renderer from "react-test-renderer"

jest.mock("../../api/auth", () => {
    return {
        auth: {
            logIn: jest.fn(),
        },
    }
})

import Login from "../login"
import { auth } from "../../api/auth"

describe("Login page", () => {
    it("renders correctly", () => {
        const loginPage = renderer.create(<Login />)

        expect(loginPage.toJSON()).toMatchSnapshot()
        expect(auth.logIn).toHaveBeenCalled()
    })
})
