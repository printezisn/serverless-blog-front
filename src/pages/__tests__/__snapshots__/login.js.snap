// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page renders correctly 1`] = `""`;
